nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/registry/reflect/MarshallerInspector.java

```diff
@@ -349,37 +360,62 @@ private Object getThreadInstance(RenderingContext ctx) {
     }
 
     /**
-     * Create a new instance of the marshaller.
+     * Create a new instance of the marshaller and inject the services.
```

Comment:
Create a new instance of the marshaller. it injects the reduired services if the marshaller is not a singleton. If  it's a singleton, it prepares the context variable to handle thread localized context. Then it injects the given ctx.